icaruscode/Timing/PMTTimingCorrectionsProvider.cxx

```diff
@@ -10,114 +10,107 @@
-    }
-    if ( getHTTPstatus(dataset) !=200 ){
-        throw cet::exception(fLogCategory)
-            << "Calibration database access failed. URL (" << url
-            << "). HTTP error status: " << getHTTPstatus(dataset) 
-            << ". HTTP error message: " << getHTTPmessage(dataset); 
-    }
+uint64_t icarusDB::PMTTimingCorrectionsProvider::RunToDatabaseTimestamp( uint32_t run ) {
 
-    return error;
+   // Run number to timestamp used in the db
+   // DBFolder.h only takes 19 digit (= timestamp in nano second),
+   // but ICARUS tables are currently using run numbers
+   // Step 1) Add 1000000000 to the run number; e.g., run XXXXX -> 10000XXXXX
+   // Step 2) Multiply 1000000000
+   uint64_t runNum = uint64_t(run);
+   uint64_t timestamp = runNum+1000000000;
+   timestamp *= 1000000000;
 
+   if( fVerbose ) mf::LogInfo(fLogCategory) << "Run " << runNum << " corrections from DB timestamp " << timestamp << std::endl;
+   
+   return timestamp;
 }
 
-
-// -----------------------------------------------------------------------------
+// -------------------------------------------------------------------------------
 
 /// Function to look up the calibration database at the table holding the pmt hardware cables corrections
 void icarusDB::PMTTimingCorrectionsProvider::ReadPMTCablesCorrections( uint32_t run ) { 
 
     // pmt_cables_delay: delays of the cables relative to trigger 
     // and reset distribution
-    const std::string name("pmt_cables_delays_data");
-    Dataset dataset;
-    int error = ConnectToDataset( name, run, dataset );
+    const std::string dbname("pmt_cables_delays_data");
+    lariov::DBFolder db(dbname, "", "", fCablesTag, true, false);
 
-    if (error) throw(std::exception());
+    bool ret = db.UpdateData( RunToDatabaseTimestamp(run) ); // select table based on run number   
+    mf::LogDebug(fLogCategory) << dbname + " corrections" << (ret? "": " not") << " updated for run " << run;
 
-    for( int row=4; row < getNtuples(dataset); row++ ) {
-
-        Tuple tuple = getTuple(dataset, row);
-
-        if( tuple != NULL ) {
-
-            unsigned int channel_id = getLongValue( tuple, 0, &error );
-            if( error ) throw std::runtime_error( "Encountered error while trying to access 'channel_id' on table " + name );
-            
-            // PPS reset correction
-            double reset_distribution_delay = getDoubleValue( tuple, 11, &error );
-            if( error ) throw std::runtime_error( "Encountered error '" + std::to_string(error) + "' while trying to access 'reset_distribution_delay' on table " + name );
-
-            // Trigger cable delay
-            double trigger_reference_delay = getDoubleValue( tuple, 10, &error );
-            if( error ) throw std::runtime_error( "Encountered error while trying to access 'trigger_reference_delay' on table " + name );
+    std::vector<unsigned int> channelList;
+    if (int res = db.GetChannelList(channelList); res != 0) {
```

Comment:
C++17 is supposed to understand that if there is a variable definition and no explicit condition, the condition should be that the conversion of the defined variable to `bool` be `true`, that would be `bool(res) == true` here, so equivalent to `res != 0`:
```suggestion
    if (int res = db.GetChannelList(channelList)) {
```
Note that I am not necessarily recommending this syntax, which is somehow less readable: I let it to your judgement (I might end up leaving it as is myself).

Reply:
I'd rather keep the expanded syntax. It would be embarassing if future me ended up not understading his own code.